parsl/jobs/simple_error_handler.py

```diff
@@ -6,12 +6,20 @@
 from parsl.jobs.states import JobStatus, JobState
 
 
-def simple_error_handler(executor: status_handling.BlockProviderExecutor, status: Dict[str, JobStatus], threshold: int):
+def simple_error_handler(executor: status_handling.BlockProviderExecutor, status: Dict[str, JobStatus], threshold: int = 3):
     (total_jobs, failed_jobs) = _count_jobs(status)
     if total_jobs >= threshold and failed_jobs == total_jobs:
         executor.set_bad_state_and_fail_all(_get_error(status))
 
 
+def windowed_error_handler(executor: status_handling.BlockProviderExecutor, status: Dict[str, JobStatus], threshold: int = 3):
+    sorted_status = [(key, status[key]) for key in sorted(status)]
```

Comment:
this sort is a bit suspicious if `status` is keyed by strings, not ints, as the type signature says... does a failing sequence of blocks 10,11,12 get ignored here because later/earlier blocks 7,8,9 were successful?

Reply:
Yep, there was a bug here, I've updated the sorted call to take a lambda that assumes the key is a simple integer, which should work for our ClusterProviders.

Reply:
that's suspicious in a different way - I guess it doesn't make sense for block IDs to be anything other than sequential integers, and maybe the rest of the codebase should be changed to not use `str` for block IDs (not in this PR).